plugin/src/main/java/org/opensearch/ml/task/MLTrainingTaskRunner.java

```diff
@@ -112,17 +113,39 @@ public void createMLTaskAndTrain(MLTrainingTaskRequest request, ActionListener<M
             .workerNode(clusterService.localNode().getId())
             .createTime(now)
             .lastUpdateTime(now)
+            .async(request.isAsync())
             .build();
 
-        mlTaskManager.createMLTask(mlTask, ActionListener.wrap(r -> {
-            mlTask.setTaskId(r.getId());
+        if (request.isAsync()) {
+            mlTaskManager.createMLTask(mlTask, ActionListener.wrap(r -> {
+                String taskId = r.getId();
+                mlTask.setTaskId(taskId);
+                if (mlTask.isAsync()) {
```

Comment:
Is there any user case that the request is Sync, but the mlTask is Async? I am just wondering the difference between the two.

Reply:
Async property of `MLTask` is consistent with request, check line 116 of this class. 
For async task, we will cache task in memory and persist task in index and it will return task id directly.
For sync task, we just cache task in memory. Will train model and return the model id.